server/src/State.ts

```diff
@@ -383,12 +389,13 @@ export class State extends Schema {
         const role = player.role;
         console.debug(`roleID=${JSON.stringify(role)}`);
 
-        switch (role.name) {
+        let roleName = role.doppelganger ? "doppelganger" : role.name;
```

Comment:
was this added because the doppelganger truly becomes a role?

Reply:
and/or what is `role.name` when `role.doppelganger` is set

Reply:
That's correct. With the current architecture, it made sense to have something to indicate the doppelganged role while still representing to the game's logic that a player is the correct role during processing. See `distributeDoppelsRole` (`State` line 251) for more information on how the doppelganged role is assigned.